ex04.ex

```diff
@@ -75,7 +77,12 @@ defmodule Ex04 do
   return value will be the thing you have to manipulate.
   """
 
-  def even_odd . . . "your code"
+  def even_odd(list) do
```

Comment:
## -1

You can remove duplication and make smaller functions by abstracting out the body of the reduce:

~~~ elixir
  def even_odd(list) do
    list
    |> reverse()
    |> reduce({ [], [] }, &partition_even_odd/2)
  end

  defp partition_even_odd(x, { es, os }) when Integer.is_even(x), do: {[x | es], os}
  defp partition_even_odd(x, { es, os }) when Integer.is_odd(x),  do: {es, [x | os]}
~~~

## +1

You're the first person to notice that if you reverse the original list, you don't have to reverse the two result lists.